src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -104,6 +104,9 @@ const FLOATING_LABEL_DEFAULT_DOCKED_TRANSFORM = `translateY(-50%)`;
     '[class.mat-form-field-disabled]': '_control.disabled',
     '[class.mat-form-field-autofilled]': '_control.autofilled',
     '[class.mat-form-field-no-animations]': '_animationMode === "NoopAnimations"',
+    '[class.mat-form-field-appearance-fill]': 'appearance == "fill"',
```

Comment:
shouldn't these all be mat-mdc-*?

Reply:
I was following the pattern set by the rest of the classses. @devversion might have more context around the naming.

Reply:
Those classes could be considered public API of the form-field, so I kept them without `mdc`. That should ensure backwards compatibility with custom controls as those could provide custom styling based on such classes.

I assume it being a trade-off. The other way around it could mean that some special overrides of the standard form-field accidentally target the MDC-based one (though that seems a bit unlikely for those classes; and rather the classes are used for the custom controls; which would be fine without the MDC prefix). I don't feel very strongly here though.

Reply:
I see, can we just leave a comment explaining why these are not mat-mdc-? It definitely looks like a mistake at first glance. We should also make sure that if we're targeting styles at any of these classes we use the unique selector (e.g. `.mat-form-field.mat-form-field-appearance-fill` or `.mat-mdc-form-field.mat-form-field-appearance-fill`)

Reply:
Yeah, I wonder though if we should expect any differences for the appearance classes. I'd expect custom controls that rely on these to not care about whether the MDC or non-MDC form-field is used.